src/rpc/blockchain.cpp

```diff
@@ -51,14 +51,14 @@ extern void TxToJSON(const CTransaction& tx, const uint256 hashBlock, UniValue&
 /* Calculate the difficulty for a given block index,
  * or the block index of the given chain.
  */
-double GetDifficulty(const CChain& chain, const CBlockIndex* blockindex)
+double GetDifficulty(const CBlockIndex* tip, const CBlockIndex* blockindex)
```

Comment:
commit: rpc: Specify chain tip instead of chain in GetDifficulty

I don't think this is really necessary. Would rather just inline `GetDifficulty(blockindex ? blockindex : chainActive.Tip())` at the few call sites if blockindex may be null.

I don't actually see any instances in this file where `blockindex` might be `nullptr`.

Reply:
Below there is the test
```cpp
// Verify that difficulty is 1.0 for a null chain tip.
BOOST_AUTO_TEST_CASE(get_difficulty_for_null_tip)
{
    double difficulty = GetDifficulty(nullptr, nullptr);
    RejectDifficultyMismatch(difficulty, 1.0);
}
```
Do you agree changing to
```cpp
// Verify that difficulty is 1.0 for a null block index.
BOOST_AUTO_TEST_CASE(get_difficulty_for_null_block)
{
    double difficulty = GetDifficulty(nullptr);
    RejectDifficultyMismatch(difficulty, 1.0);
}
```
or should change the return value?

If you don't mind I would rather keep the existing behaviour and discuss this elsewhere.

Reply:
Yes, I am proposing changing the behavior so that if null is passed it, it returns 1. I think that makes the most sense in the context of this change because you want are removing the dependence of this function on `chainActive`. Given that you are updating every invocation, this seems like the right time to make the change I am proposing.

Reply:
Fixed in 92e6839.